### ISSUE TITLE
Gamma Function has nonsensical behavior with large arguments

### ISSUE THREAD

**Issue by mscotch1** [jstat/jstat#250]
## Description
If `jStat.gammafn` is passed a large number, it can take a long time to return a value, and the value may be nonsensical.

e.g.
```js
jStat.gammafn(1e16); // takes ~2.5 seconds to return Infinity
jStat.gammafn(1e18); // returns 5.56e-2
```

## Demo
https://jsfiddle.net/uovwja0z/1/

**Comment by trevnorris** [comment]
Would something like this work for you?
```diff
@@ -72,6 +72,9 @@ jStat.gammafn = function gammafn(x) {
   var xnum = 0;
   var y = x;
   var i, z, yi, res;
+  if (x > 171.6243769536076) {
+    return Infinity;
+  }
   if (y <= 0) {
     res = y % 1 + 3.6e-16;
     if (res) {
```
It immediately checks for the maximum non infinite value, and returns immediately if it's greater than that.

Also, it appears that `gammafn()` fails with negative numbers, but I'm not sure what the expected behavior of that should be.

**Comment by mscotch1** [comment]
Yes, I think that would sufficiently address the issues we've run into with the behavior of `gammafn`. And we don't need this behavior for negative numbers, but I would expect `gammafn` to be able to calculate the gamma function for any number in its domain (see https://en.wikipedia.org/wiki/Gamma_function), so all negative non-integers.

**Comment by trevnorris** [comment]
Seems I made the change but never pushed it. The fix landed in 25d3cf40f.
